--- ./usr/sbin/start_pms.orig	2020-01-17 05:17:10.000000000 +0100
+++ ./usr/sbin/start_pms	2020-01-21 20:05:07.772901873 +0100
@@ -29,6 +29,9 @@
   export PLEX_MEDIA_SERVER_USER="$PLEX_USER"
 fi
 
+# Activate python virtualenv
+. "${PLEX_MEDIA_SERVER_HOME}"/Resources/Python/bin/activate
+
 # Create AppSuppDir if not present
 if [ ! -d "$PLEX_MEDIA_SERVER_APPLICATION_SUPPORT_DIR" ]; then
   mkdir -p "$PLEX_MEDIA_SERVER_APPLICATION_SUPPORT_DIR"
