--- encfs/encfsctl.pod.orig	2014-09-06 11:48:22.217311534 +0200
+++ encfs/encfsctl.pod	2014-09-06 11:49:36.271276162 +0200
@@ -1,4 +1,7 @@
+=pod
+
 =cut
+
 Copyright (c) 2003-2004, Valient Gough <vgough@pobox.com>
 All rights reserved.
 
@@ -7,8 +10,6 @@
 Foundation; either version 2 of the License, or (at your option) any later
 version.
 
-=pod
-
 =head1 NAME
 
 encfsctl - administrative tool for working with EncFS filesystems
