--- ./Makefile.boot.orig	2018-02-18 20:59:04.008980556 +0100
+++ ./Makefile.boot	2018-02-18 20:58:47.839602062 +0100
@@ -10,8 +10,8 @@
 CFLAGS=	$(OPT) $(LARGE_CFLAGS) $(LOCAL_CFLAGS)
 
 GROFF=	groff -Tascii
-CRYPTOLIBDIR=	# -L/usr/local/lib
-CRYPTOLIBS=	$(CRYPTOLIBDIR) -lcrypto -lssl
+CRYPTOLIBDIR=	# -L/usr/lib
+CRYPTOLIBS=	$(CRYPTOLIBDIR) -lcrypt -lssl -lcrypto
 
 FILES=	bozohttpd.c auth-bozo.c cgi-bozo.c content-bozo.c daemon-bozo.c \
 	dir-index-bozo.c lua-bozo.c ssl-bozo.c tilde-luzah-bozo.c main.c
