$NetBSD: patch-ab,v 1.20 2014/01/02 09:02:22 mrg Exp $

--- ./Makefile.boot.orig	2014-01-01 23:59:54.000000000 -0800
+++ ./Makefile.boot	2014-01-02 00:02:15.000000000 -0800
@@ -5,7 +5,7 @@
 
 CC=	cc
-OPT=	-O
+OPT=
-LARGE_CFLAGS=	-D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64
+LARGE_CFLAGS=
 LOCAL_CFLAGS=	-DNO_LUA_SUPPORT
 CFLAGS=	$(OPT) $(LARGE_CFLAGS) $(LOCAL_CFLAGS)
 
$NetBSD: patch-bozohttpd.c,v 1.4 2017/02/05 13:32:16 leot Exp $

Adjust SERVER_SOFTWARE to intended version.

From -r1.86 of NetBSD's src/libexec/httpd/bozohttpd.c

--- ./bozohttpd.c.orig	2017-02-01 04:21:23.000000000 +0000
+++ ./bozohttpd.c
@@ -109,7 +109,7 @@
 #define INDEX_HTML		"index.html"
 #endif
 #ifndef SERVER_SOFTWARE
-#define SERVER_SOFTWARE		"bozohttpd/20170102"
+#define SERVER_SOFTWARE		"bozohttpd/20170201"
 #endif
 #ifndef DIRECT_ACCESS_FILE
 #define DIRECT_ACCESS_FILE	".bzdirect"
