--- lightning-23.05.1/wallet/wallet.h.orig	2023-06-06 04:31:42.000000000 +0200
+++ lightning-23.05.1/wallet/wallet.h	2023-06-11 08:21:07.936948968 +0200
@@ -406,6 +406,12 @@
 
 	/* Fully parsed transaction */
 	const struct bitcoin_tx *tx;
+
+	/* tal_arr containing the annotation types, if any, for the respective
+	 * inputs and outputs. 0 if there are no annotations for the
+	 * element. */
+	struct tx_annotation *input_annotations;
+	struct tx_annotation *output_annotations;
 };
 
 /**
