--- Makefile.orig	2016-08-01 07:51:03.205597457 +0200
+++ Makefile	2016-08-01 10:40:14.710293381 +0200
@@ -1,6 +1,8 @@
 CC	?= gcc
-CFLAGS	?= -O2 -Wall -Wextra -Wno-missing-field-initializers -Wno-unused-parameter -g2
-LDFLAGS ?= -lrt
+CFLAGS ?= -O2 -g2
+CFLAGS += -Wall -Wextra -Wno-missing-field-initializers -Wno-unused-parameter
+
+LDLIBS += -lrt
 
 all: flashbench erase
 
@@ -9,10 +11,15 @@
 flashbench.o: flashbench.c vm.h dev.h
 
 flashbench: flashbench.o dev.o vm.o
-	$(CC) -o $@ flashbench.o dev.o vm.o $(LDFLAGS)
-
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ $(LDLIBS)
 
 erase: erase.o
 
+install:	all
+	mkdir -p $(DESTDIR)/usr/bin
+	install -m 755 -D flashbench erase $(DESTDIR)/usr/bin/
+
 clean:
 	rm -f flashbench flashbench.o erase erase.o dev.o vm.o
+
+.PHONY: all clean
