--- ./kernel/sysctl.c.orig	2019-05-04 19:12:52.568113654 +0200
+++ ./kernel/sysctl.c	2019-05-04 19:31:53.935312505 +0200
@@ -130,6 +130,7 @@
 static int __maybe_unused four = 4;
 static unsigned long zero_ul;
 static unsigned long one_ul = 1;
+static unsigned long long_max = LONG_MAX;
 static int __read_mostly one_hundred = 100;
 static int __read_mostly one_thousand = 1000;
 #ifdef CONFIG_SCHED_PDS
