--- b/src/app/qsci/lexers/qscilexerada.h
+++ a/src/app/qsci/lexers/qscilexerada.h
@@ -3,6 +3,9 @@
 
 #include <Qsci/qscilexer.h>
 
+// located in SciLexer.h
+#define SCLEX_ADA 20
+
 #define SCE_ADA_DEFAULT 0
 #define SCE_ADA_WORD 1
 #define SCE_ADA_IDENTIFIER 2
--- b/src/app/qsci/lexers/qscilexerhaskell.h
+++ a/src/app/qsci/lexers/qscilexerhaskell.h
@@ -3,6 +3,9 @@
 
 #include <Qsci/qscilexer.h>
 
+// located in SciLexer.h
+#define SCLEX_HASKELL 68
+
 #define SCE_HA_DEFAULT 0
 #define SCE_HA_IDENTIFIER 1
 #define SCE_HA_KEYWORD 2
--- b/src/app/qsci/lexers/qscilexerlisp.h
+++ a/src/app/qsci/lexers/qscilexerlisp.h
@@ -3,6 +3,9 @@
 
 #include <Qsci/qscilexer.h>
 
+// located in SciLexer.h
+#define SCLEX_LISP 21
+
 #define SCE_LISP_DEFAULT 0
 #define SCE_LISP_COMMENT 1
 #define SCE_LISP_NUMBER 2
--- b/src/app/qsci/lexers/qscilexernsis.h
+++ a/src/app/qsci/lexers/qscilexernsis.h
@@ -3,6 +3,9 @@
 
 #include <Qsci/qscilexer.h>
 
+// located in SciLexer.h
+#define SCLEX_NSIS 43
+
 #define SCE_NSIS_DEFAULT 0
 #define SCE_NSIS_COMMENT 1
 #define SCE_NSIS_STRINGDQ 2
