diff -ur --unidirectional-new-file -ur juffed-9999.orig/CMakeLists.txt juffed-9999/CMakeLists.txt
--- juffed-9999.orig/CMakeLists.txt	2016-02-04 13:34:41.891954151 +0000
+++ juffed-9999/CMakeLists.txt	2016-02-04 13:35:56.030952182 +0000
@@ -20,7 +20,6 @@
 cmake_minimum_required (VERSION 2.8.11)
 
 # build options
-option(USE_QT5 "Use Qt5. Defaults to Qt4" $ENV{USE_QT5})
 option(USE_SYSTEM_QTSINGLEAPPLICATION "Try to find system-wide QtSingleApplication library" $ENV{USE_SYSTEM_SINGLETON})
 option(USE_ENCA "Try to find Enca library for file encoding check" $ENV{USE_ENCA})
 
@@ -92,22 +91,6 @@
 
 set (CMAKE_AUTOMOC ON)
 
-if (USE_QT5)
-    find_package ( Qt5 COMPONENTS Widgets Xml Network PrintSupport LinguistTools CONFIG REQUIRED )
-    set (JUFF_QT_IMPORTED_TARGETS Qt5::Widgets Qt5::Xml Qt5::Network Qt5::PrintSupport)
-
-    macro(qt_wrap_ui)
-      qt5_wrap_ui(${ARGN})
-    endmacro()
-
-    macro(qt_add_resources)
-      qt5_add_resources(${ARGN})
-    endmacro()
-
-    macro(qt_add_translation)
-      qt5_add_translation(${ARGN})
-    endmacro()
-else (USE_QT5)
     find_package ( Qt4 COMPONENTS QtCore QtGui QtXml QtNetwork REQUIRED )
     set (JUFF_QT_IMPORTED_TARGETS Qt4::QtCore Qt4::QtGui Qt4::QtXml Qt4::QtNetwork)
 
@@ -122,7 +105,6 @@
     macro(qt_add_translation)
       qt4_add_translation(${ARGN})
     endmacro()
-endif (USE_QT5)
 
 # this is mandatory for Apple builds to load plugin correctly
 add_definitions(-DQT_NO_DEBUG)
