--- ./src/GNUmakefile.orig	2023-04-23 12:46:34.402669147 +0200
+++ ./src/GNUmakefile	2023-04-23 12:49:08.580687473 +0200
@@ -21,7 +21,7 @@
 HAS_PYTHON_CONFIG := $(shell ${PYTHON}-config --ldflags ${PYTHON_EMBED} 2>/dev/null)
 
 CPPFLAGS += -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -I. -I$(VPATH) -I$(VPATH)/../lib/uthash/include
-CFLAGS   += -g -Wall -Wextra -O2
+CFLAGS   += -Wall -Wextra
 CFLAGS   += $(shell pkg-config --cflags talloc)
 LDFLAGS  += -Wl,-z,noexecstack
 LDFLAGS  += $(shell pkg-config --libs talloc)
