diff --git a/src/GNUmakefile b/src/GNUmakefile
index 38d89a9..b848a3b 100644
--- a/src/GNUmakefile
+++ b/src/GNUmakefile
@@ -21,7 +21,7 @@ PYTHON_EMBED = $(shell ${PYTHON} -c "import sys; print('--embed' if sys.hexversi
 HAS_PYTHON_CONFIG := $(shell ${PYTHON}-config --ldflags ${PYTHON_EMBED} 2>/dev/null)
 
 CPPFLAGS += -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -I. -I$(VPATH)
-CFLAGS   += -g -Wall -Wextra -O2
+CFLAGS   += -Wall -Wextra
 CFLAGS   += $(shell pkg-config --cflags talloc)
 LDFLAGS  += -Wl,-z,noexecstack
 LDFLAGS  += $(shell pkg-config --libs talloc)
