--- ./src/GNUmakefile	2026-09-12 09:00:17.755263301 +0200
+++ ./src/GNUmakefile	2026-09-12 09:01:34.107848834 +0200
@@ -57,8 +57,10 @@
 endif
 
 CFLAGS   += $(shell ${PKG_CONFIG} --cflags talloc)
+CFLAGS   += $(shell ${PYTHON}-config --cflags)
 LDFLAGS  += -Wl,-z,noexecstack
 LDFLAGS  += $(shell ${PKG_CONFIG} --libs talloc)
+LDFLAGS  += $(shell ${PYTHON}-config --embed --ldflags)
 
 ifneq ($(findstring -static,$(LDFLAGS)),)
 CARE_LDFLAGS  = -Wl,--start-group $(shell ${PKG_CONFIG} --libs --static libarchive) \
