diff --git a/src/GNUmakefile b/src/GNUmakefile
index 38d89a9..65b94b8 100644
--- a/src/GNUmakefile
+++ b/src/GNUmakefile
@@ -23,8 +23,10 @@ HAS_PYTHON_CONFIG := $(shell ${PYTHON}-config --ldflags ${PYTHON_EMBED} 2>/dev/n
 CPPFLAGS += -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -I. -I$(VPATH)
 CFLAGS   += -g -Wall -Wextra -O2
 CFLAGS   += $(shell pkg-config --cflags talloc)
+CFLAGS   += $(shell ${PYTHON}-config --cflags)
 LDFLAGS  += -Wl,-z,noexecstack
 LDFLAGS  += $(shell pkg-config --libs talloc)
+LDFLAGS  += $(shell ${PYTHON}-config --embed --ldflags)
 
 CARE_LDFLAGS  = $(shell pkg-config --libs libarchive)
 
