--- spotifyblob/blob/CMakeLists.txt
+++ spotifyblob/blob/CMakeLists.txt
@@ -70,7 +70,7 @@
 
   install(
     FILES ${CMAKE_BINARY_DIR}/clementine-spotifyblob
-    DESTINATION ${CMAKE_BINARY_DIR}/spotify/version${SPOTIFY_BLOB_VERSION}-${SPOTIFY_BLOB_ARCH}bit/
+    DESTINATION ${CMAKE_INSTALL_PREFIX}/share/clementine/spotify/version${SPOTIFY_BLOB_VERSION}-${SPOTIFY_BLOB_ARCH}bit/
     RENAME blob
   )
 endif(LINUX)
