--- ./dmd/mars.c.old	2012-12-31 21:16:46.000000000 +0100
+++ ./dmd/mars.c	2018-01-25 10:21:57.345041141 +0100
@@ -506,7 +506,7 @@
 #if _WIN32
     inifilename = inifile(argv[0], "sc.ini", "Environment");
 #elif linux || __APPLE__ || __FreeBSD__ || __OpenBSD__ || __sun
-    inifilename = inifile(argv[0], "dmd.conf", "Environment");
+    inifilename = inifile(argv[0], "dmd1.conf", "Environment");
 #else
 #error "fix this"
 #endif
